citest/json_predicate/map_predicate.py

```diff
@@ -153,6 +153,21 @@ def __eq__(self, result):
             and self.__good_map == result.good_object_result_mappings
             and self.__bad_map == result.bad_object_result_mappings)
 
+  def clone_with_source(self, source, base_target_path, base_value_path):
+    """Implements CloneableWithNewSource interface."""
+    builder = MapPredicateResultBuilder(self.pred)
+
+    for index, orig in enumerate(self.results):
+      if isinstance(orig, predicate.CloneableWithNewSource):
```

Comment:
For instance, in Java there's `implements ...` where you expect certain functions to be present. Is this Python's version of the same thing?